<ui version="4.0" >
 <class>TrajectoryGraphicsWindow</class>
 <widget class="QWidget" name="TrajectoryGraphicsWindow" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1028</width>
    <height>825</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" >
   <item>
    <widget class="QTextEdit" name="glPanePlaceholderTextEdit" >
     <property name="sizePolicy" >
      <sizepolicy vsizetype="Expanding" hsizetype="Expanding" >
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="controlsFrame" >
     <property name="sizePolicy" >
      <sizepolicy vsizetype="Fixed" hsizetype="MinimumExpanding" >
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="frameShape" >
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow" >
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" >
      <item>
       <widget class="QGroupBox" name="simulationGroupBox" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Preferred" hsizetype="Preferred" >
          <horstretch>1</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title" >
         <string>Simulation</string>
        </property>
        <layout class="QVBoxLayout" >
         <item>
          <widget class="QProgressBar" name="simulationProgressBar" >
           <property name="value" >
            <number>0</number>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" >
           <item>
            <widget class="QLabel" name="statusSimulationLabel" >
             <property name="sizePolicy" >
              <sizepolicy vsizetype="Fixed" hsizetype="Preferred" >
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text" >
              <string>Status:</string>
             </property>
             <property name="buddy" >
              <cstring>statusSimulationLineEdit</cstring>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="statusSimulationLineEdit" >
             <property name="text" >
              <string>Finished</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPushButton" name="abortSimulationButton" >
           <property name="sizePolicy" >
            <sizepolicy vsizetype="Expanding" hsizetype="Expanding" >
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text" >
            <string>Abort</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="visualizationgroupBox" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Preferred" hsizetype="Minimum" >
          <horstretch>3</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title" >
         <string>Visualization</string>
        </property>
        <layout class="QVBoxLayout" >
         <item>
          <layout class="QHBoxLayout" >
           <item>
            <layout class="QVBoxLayout" >
             <item>
              <layout class="QHBoxLayout" >
               <item>
                <widget class="QLabel" name="numFramesLabel" >
                 <property name="text" >
                  <string>Number of frames:</string>
                 </property>
                 <property name="buddy" >
                  <cstring>numFramesLineEdit</cstring>
                 </property>
                </widget>
               </item>
               <item>
                <spacer>
                 <property name="orientation" >
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" >
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item>
                <widget class="QLineEdit" name="numFramesLineEdit" >
                 <property name="sizePolicy" >
                  <sizepolicy vsizetype="Fixed" hsizetype="Fixed" >
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="readOnly" >
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" >
               <item>
                <widget class="QLabel" name="playbackSpeedLabel" >
                 <property name="text" >
                  <string>% playback speed:</string>
                 </property>
                 <property name="buddy" >
                  <cstring>playbackSpeedSpinBox</cstring>
                 </property>
                </widget>
               </item>
               <item>
                <spacer>
                 <property name="orientation" >
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" >
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item>
                <widget class="QSpinBox" name="playbackSpeedSpinBox" >
                 <property name="sizePolicy" >
                  <sizepolicy vsizetype="Fixed" hsizetype="Fixed" >
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="minimum" >
                  <number>10</number>
                 </property>
                 <property name="maximum" >
                  <number>800</number>
                 </property>
                 <property name="singleStep" >
                  <number>10</number>
                 </property>
                 <property name="value" >
                  <number>100</number>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item>
            <widget class="Line" name="line_4" >
             <property name="sizePolicy" >
              <sizepolicy vsizetype="Expanding" hsizetype="Fixed" >
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="orientation" >
              <enum>Qt::Vertical</enum>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" >
             <item>
              <layout class="QVBoxLayout" >
               <item>
                <widget class="QLabel" name="beginFrameLabel" >
                 <property name="text" >
                  <string>Begin</string>
                 </property>
                 <property name="buddy" >
                  <cstring>beginFrameSpinBox</cstring>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="beginFrameSpinBox" >
                 <property name="sizePolicy" >
                  <sizepolicy vsizetype="Fixed" hsizetype="Minimum" >
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="maximum" >
                  <number>1000000</number>
                 </property>
                 <property name="value" >
                  <number>1000000</number>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QVBoxLayout" >
               <item>
                <layout class="QHBoxLayout" >
                 <item>
                  <spacer>
                   <property name="orientation" >
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeHint" >
                    <size>
                     <width>61</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                 <item>
                  <widget class="QLabel" name="currentFrameLabel" >
                   <property name="text" >
                    <string>Frame:</string>
                   </property>
                   <property name="buddy" >
                    <cstring>currentFrameSpinBox</cstring>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QSpinBox" name="currentFrameSpinBox" >
                   <property name="sizePolicy" >
                    <sizepolicy vsizetype="Fixed" hsizetype="Minimum" >
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="maximum" >
                    <number>1000000</number>
                   </property>
                   <property name="value" >
                    <number>1000000</number>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <spacer>
                   <property name="orientation" >
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeHint" >
                    <size>
                     <width>61</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                </layout>
               </item>
               <item>
                <widget class="QSlider" name="currentFrameHSlider" >
                 <property name="orientation" >
                  <enum>Qt::Horizontal</enum>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QVBoxLayout" >
               <item>
                <widget class="QLabel" name="endFrameLabel" >
                 <property name="text" >
                  <string>End</string>
                 </property>
                 <property name="buddy" >
                  <cstring>endFrameSpinBox</cstring>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="endFrameSpinBox" >
                 <property name="sizePolicy" >
                  <sizepolicy vsizetype="Fixed" hsizetype="Minimum" >
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="maximum" >
                  <number>1000000</number>
                 </property>
                 <property name="value" >
                  <number>1000000</number>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <widget class="Line" name="visualizationHLine" >
           <property name="orientation" >
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" >
           <item>
            <widget class="QGroupBox" name="repetitionGroupBox" >
             <property name="sizePolicy" >
              <sizepolicy vsizetype="Preferred" hsizetype="Preferred" >
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="title" >
              <string>Repetition</string>
             </property>
             <layout class="QVBoxLayout" >
              <item>
               <widget class="QStackedWidget" name="stackedWidget_2" >
                <widget class="QWidget" name="page_3" >
                 <layout class="QHBoxLayout" >
                  <item>
                   <widget class="QRadioButton" name="noneRepetitionRadioButton" >
                    <property name="text" >
                     <string>None</string>
                    </property>
                    <property name="checked" >
                     <bool>true</bool>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QRadioButton" name="loopRepetitionRadioButton" >
                    <property name="text" >
                     <string>Loop</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QRadioButton" name="oscillateRepetitionRadioButton" >
                    <property name="text" >
                     <string>Oscillate</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
                <widget class="QWidget" name="page_4" />
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="playbackGroupBox" >
             <property name="sizePolicy" >
              <sizepolicy vsizetype="Preferred" hsizetype="Preferred" >
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="title" >
              <string>Playback</string>
             </property>
             <layout class="QHBoxLayout" >
              <item>
               <widget class="QStackedWidget" name="trajectoryFirstButtonSizingStackedWidget" >
                <property name="currentIndex" >
                 <number>0</number>
                </property>
                <widget class="QWidget" name="page" >
                 <layout class="QHBoxLayout" >
                  <item>
                   <widget class="QPushButton" name="trajectoryFirstButton" >
                    <property name="toolTip" >
                     <string>First frame</string>
                    </property>
                    <property name="statusTip" >
                     <string>Go to first frame</string>
                    </property>
                    <property name="whatsThis" >
                     <string>Go to first frame</string>
                    </property>
                    <property name="text" >
                     <string/>
                    </property>
                    <property name="icon" >
                     <iconset resource="application.qrc" >:/Icons/trajectory_first.png</iconset>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
                <widget class="QWidget" name="page_7" />
               </widget>
              </item>
              <item>
               <widget class="QStackedWidget" name="playbackSizingStackedWidget" >
                <property name="sizePolicy" >
                 <sizepolicy vsizetype="Fixed" hsizetype="Expanding" >
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="currentIndex" >
                 <number>0</number>
                </property>
                <widget class="QWidget" name="autoFrameAdvancePage" >
                 <widget class="QPushButton" name="trajectoryPlayRevButton" >
                  <property name="geometry" >
                   <rect>
                    <x>9</x>
                    <y>9</y>
                    <width>30</width>
                    <height>26</height>
                   </rect>
                  </property>
                  <property name="toolTip" >
                   <string>Play reverse</string>
                  </property>
                  <property name="statusTip" >
                   <string>Begins playing in reverse order of frames</string>
                  </property>
                  <property name="whatsThis" >
                   <string>Begins playing in reverse order of frames</string>
                  </property>
                  <property name="text" >
                   <string/>
                  </property>
                  <property name="icon" >
                   <iconset resource="application.qrc" >:/Icons/trajectory_play_rev.png</iconset>
                  </property>
                  <property name="checkable" >
                   <bool>true</bool>
                  </property>
                 </widget>
                 <widget class="QPushButton" name="trajectoryStopButton" >
                  <property name="geometry" >
                   <rect>
                    <x>49</x>
                    <y>9</y>
                    <width>71</width>
                    <height>26</height>
                   </rect>
                  </property>
                  <property name="text" >
                   <string/>
                  </property>
                  <property name="icon" >
                   <iconset resource="application.qrc" >:/Icons/trajectory_stop.png</iconset>
                  </property>
                 </widget>
                 <widget class="QPushButton" name="trajectoryPlayButton" >
                  <property name="geometry" >
                   <rect>
                    <x>129</x>
                    <y>9</y>
                    <width>30</width>
                    <height>26</height>
                   </rect>
                  </property>
                  <property name="toolTip" >
                   <string>Play forward</string>
                  </property>
                  <property name="statusTip" >
                   <string>Begin playing forward</string>
                  </property>
                  <property name="whatsThis" >
                   <string>Begins playing forward</string>
                  </property>
                  <property name="text" >
                   <string/>
                  </property>
                  <property name="icon" >
                   <iconset resource="application.qrc" >:/Icons/trajectory_play.png</iconset>
                  </property>
                  <property name="checkable" >
                   <bool>true</bool>
                  </property>
                 </widget>
                </widget>
                <widget class="QWidget" name="manualFrameAdvancePage" >
                 <layout class="QHBoxLayout" >
                  <item>
                   <widget class="QPushButton" name="trajectoryMinusFiveButton" >
                    <property name="text" >
                     <string/>
                    </property>
                    <property name="icon" >
                     <iconset resource="application.qrc" >:/Icons/trajectory_minusfive.png</iconset>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QPushButton" name="trajectoryMinusOneButton" >
                    <property name="text" >
                     <string/>
                    </property>
                    <property name="icon" >
                     <iconset resource="application.qrc" >:/Icons/trajectory_minusone.png</iconset>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QPushButton" name="trajectoryPlusOneButton" >
                    <property name="text" >
                     <string/>
                    </property>
                    <property name="icon" >
                     <iconset resource="application.qrc" >:/Icons/trajectory_plusone.png</iconset>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QPushButton" name="trajectoryPlusFiveButton" >
                    <property name="text" >
                     <string/>
                    </property>
                    <property name="icon" >
                     <iconset resource="application.qrc" >:/Icons/trajectory_plusfive.png</iconset>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </widget>
              </item>
              <item>
               <widget class="QStackedWidget" name="trajectoryLastButtonSizingStackedWidget" >
                <property name="currentIndex" >
                 <number>0</number>
                </property>
                <widget class="QWidget" name="page_10" >
                 <layout class="QHBoxLayout" >
                  <item>
                   <widget class="QPushButton" name="trajectoryLastButton" >
                    <property name="toolTip" >
                     <string>Last frame</string>
                    </property>
                    <property name="statusTip" >
                     <string>Go to last frame</string>
                    </property>
                    <property name="whatsThis" >
                     <string>Go to last frame</string>
                    </property>
                    <property name="text" >
                     <string/>
                    </property>
                    <property name="icon" >
                     <iconset resource="application.qrc" >:/Icons/trajectory_last.png</iconset>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
                <widget class="QWidget" name="page_11" />
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="frameAdvanceGroupBox" >
             <property name="sizePolicy" >
              <sizepolicy vsizetype="Preferred" hsizetype="Preferred" >
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="title" >
              <string>Frame Advance</string>
             </property>
             <layout class="QHBoxLayout" >
              <item>
               <widget class="QStackedWidget" name="stackedWidget_3" >
                <widget class="QWidget" name="page_5" >
                 <layout class="QHBoxLayout" >
                  <item>
                   <widget class="QRadioButton" name="autoFrameAdvanceRadioButton" >
                    <property name="text" >
                     <string>Auto</string>
                    </property>
                    <property name="checked" >
                     <bool>true</bool>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QRadioButton" name="manualFrameAdvanceRadioButton" >
                    <property name="text" >
                     <string>Manual</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
                <widget class="QWidget" name="page_6" />
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="application.qrc" />
 </resources>
 <connections>
  <connection>
   <sender>currentFrameSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>currentFrameHSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>796</x>
     <y>648</y>
    </hint>
    <hint type="destinationlabel" >
     <x>842</x>
     <y>680</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>currentFrameHSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>currentFrameSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>754</x>
     <y>679</y>
    </hint>
    <hint type="destinationlabel" >
     <x>769</x>
     <y>660</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>trajectoryPlusOneButton</sender>
   <signal>clicked()</signal>
   <receiver>currentFrameSpinBox</receiver>
   <slot>stepUp()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>656</x>
     <y>759</y>
    </hint>
    <hint type="destinationlabel" >
     <x>814</x>
     <y>645</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>trajectoryMinusOneButton</sender>
   <signal>clicked()</signal>
   <receiver>currentFrameSpinBox</receiver>
   <slot>stepDown()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>618</x>
     <y>753</y>
    </hint>
    <hint type="destinationlabel" >
     <x>821</x>
     <y>648</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
